### PR TITLE
Add trivial check for preset operation in chroot

### PR DESCRIPTION
**Pull request by keszybz**
~~This is the middle commit from #25616 and a very simple test case to check that it works.~~

### CODE REVIEW

**Comment by keszybz** [comment]
CIFuzz / Fuzzing (undefined, x86_64) is finding a failure in fuzz-dsn-resource-record.

**Comment by poettering** [comment]
so what's the strategy then to get your chroot environments fixed?

File: test/units/testsuite-65.sh [hunk 1]
```diff
@@ -139,6 +139,14 @@ systemd-analyze cat-config systemd/system.conf systemd/journald.conf >/dev/null
 systemd-analyze cat-config systemd/system.conf foo/bar systemd/journald.conf >/dev/null
 systemd-analyze cat-config foo/bar
 
+# check that systemd-analyze cat-config paths work in a chroot
+mkdir -p /tmp/root
+mount --bind / /tmp/root
+systemd-analyze cat-config systemd/system-preset >/tmp/out1
+chroot /tmp/root systemd-analyze cat-config systemd/system-preset >/tmp/out2
+diff /tmp/out{1,2}
```
Comment (mrc0mmand) [inline test/units/testsuite-65.sh]: I don't think we can make ASan work with the chroot stuff (at least in some reasonable way), so could you please just wrap this block in:

```suggestion
if [[ -z "${ASAN_OPTIONS:-}" ]]; then
    # check that systemd-analyze cat-config paths work in a chroot
    mkdir -p /tmp/root
    mount --bind / /tmp/root
    systemd-analyze cat-config systemd/system-preset >/tmp/out1
    chroot /tmp/root systemd-analyze cat-config systemd/system-preset >/tmp/out2
    diff /tmp/out{1,2}
fi
```

to skip the test when running with sanitizers?

Comment (keszybz) [reply test/units/testsuite-65.sh]: Thanks.

File: test/units/testsuite-65.sh [hunk 2]
```diff
@@ -139,6 +139,16 @@ systemd-analyze cat-config systemd/system.conf systemd/journald.conf >/dev/null
 systemd-analyze cat-config systemd/system.conf foo/bar systemd/journald.conf >/dev/null
 systemd-analyze cat-config foo/bar
 
+if [[ -v ASAN_OPTIONS ]]; then
```
Comment (mrc0mmand) [inline test/units/testsuite-65.sh]: Since you went with `-v`, you'll need to invert the condition:

```suggestion
if [[ ! -v ASAN_OPTIONS ]]; then
```

Comment (keszybz) [reply test/units/testsuite-65.sh]: Duh.

**Review by bluca — APPROVED** [review]

**Pull request merged**